teps/0142-enable-step-reusability.md

```diff
@@ -0,0 +1,997 @@
+
+![workspace overhead](images/0142-workspace-overhead.png)
+
+In the case where `Pods` are scheduled to different nodes in a multi-node cluster, there’s an additional overhead of ~10s for node reattachment besides the ~10s of mounting a `Persistent Volume`. Users can schedule `Pods` to different nodes by disabling `Affinity Assistant`.
+
+![multi node workspace overhead](images/0142-multi-node-workspace-overhead.png)
+
+### Security
+
+Kubernetes supports two types of volumes: [Ephemeral Volumes](https://kubernetes.io/docs/concepts/storage/ephemeral-volumes/) which are tied to the lifetime of a `Pod`, and [Persistent Volumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) which exist beyond the lifetime of a `Pod`. Given that reusable units in Tekton (i.e. `Tasks`) execute in separate `Pods`, they can only natively use `Persistent Volumes` to share data. This limits users who would prefer to use `Ephemeral Volumes` to share data.
+
+A user [noted](https://github.com/tektoncd/pipeline/issues/6601) that: “The ability to modify `Persistent Volumes` contents is an additional threat. If the contents are changed between a git clone `Task` and a container build `Task`, there is no longer a guarantee the build is of the git reference that was checked out. The ability to change the contents of a `Persistent Volume` is granted to anyone that can create a `Pod`”.
+ 
+Enabling use of `Ephemeral Volumes` to share data between reusable units of work via local disks will limit external influence, thereby improving the security of workloads. 
+
+### Case Study
+
+A user relies on [git](https://github.com/tektoncd/catalog/tree/main/task/git-clone/0.9) `Task` to fetch source code from a repo and [kaniko](https://github.com/tektoncd/catalog/blob/main/task/kaniko/0.6/kaniko.yaml) `Task` to build and push an image. 
+
+If they copy and paste the `Steps` of the `Tasks` into a single `Task` to execute in one `Pod`, then the source code will be fetched to a local disk which will be used to build the image. This is performant but hurts reusability.
+
+![clone build push task](images/0142-clone-build-push-task.png)
+
+If they execute the `Tasks` in separate `Pods`, the source code will be fetched to a persistent disk which will be used to build the image. This maintains the well-factored reuse but incurs performance costs.
+
+![clone build push pipeline](images/0142-clone-build-push-pipeline.png)
+
+### Use Cases
+
+1. As a Platform Engineer, I want to take the work that the `Steps` in my `Tasks` perform and make them easily usable and reusable by other teams.
+2. As a DevOps Engineer, I want to optimize the performance of Tekton workloads by reducing the execution time and resource utilization.
+3. As a Security Engineer, I want developers to use ephemeral local volumes to share data between reusable units of work to limit external influence workloads. 
+
```

Comment:
Also, sometimes single step `Task` hinders the reusability for QE.
This use-case involves requiring any scaffolding or services to fully run E2E. 
```
We need to manually define these services using sidecars in Tekton, such as:


sidecars:
  - image: $(params.IMAGE)
     name: postgres
     script: |
       #! /bin/bash
       source /home/postgres/run_postgres.sh;
  - image: $(params.IMAGE)
     name: keycloak
  #etc...
```

But we need to do that for different e2e test Tasks. The alternative is to write multiple steps which reduces the usability of Tasks.